Objects/genobject.c

```diff
@@ -327,23 +328,37 @@ gen_close_iter(PyObject *yf)
 }
 
 static inline bool
-is_resume(_Py_CODEUNIT *instr)
+is_resume(_PyInterpreterFrame *frame, uint8_t *oparg_p)
 {
-    uint8_t code = FT_ATOMIC_LOAD_UINT8_RELAXED(instr->op.code);
-    return (
-        code == RESUME ||
-        code == RESUME_CHECK ||
-        code == INSTRUMENTED_RESUME
-    );
+    PyCodeObject *code = _PyFrame_GetCode(frame);
+    int offset = frame->instr_ptr - _PyCode_CODE(code);
+    uint8_t opcode = _Py_GetBaseOpcode(code, offset);
+    uint8_t oparg = frame->instr_ptr->op.arg;
+#ifdef _Py_TIER2
+    if (opcode == ENTER_EXECUTOR) {
+        _PyExecutorObject *executor = _Py_GetExecutor(code, sizeof(_Py_CODEUNIT) * offset);
+        opcode = _PyOpcode_Deopt[executor->vm_data.opcode];
+        oparg = executor->vm_data.oparg;
+        Py_DECREF(executor);
+    }
+#endif
+    if (opcode == RESUME) {
+        *oparg_p = oparg;
+        return true;
+    }
+    return false;
 }
 
 PyObject *
 _PyGen_yf(PyGenObject *gen)
 {
     if (gen->gi_frame_state == FRAME_SUSPENDED_YIELD_FROM) {
         _PyInterpreterFrame *frame = &gen->gi_iframe;
-        assert(is_resume(frame->instr_ptr));
```

Comment:
The new implementation of `is_resume` is a bit strange, with it returning the `oparg` via an output parameter.
I think we need two functions. One that *quickly* returns yes/probably not for the gen_close and another one for sanity checking that can be slow. For that we will want a better API for getting the original instruction.

That's all too much for this PR though. So, can you go back to to just dropping the asserts?
I've made an for sorting out `is_resume` and a better API for getting the original instruction.